src/read/elf/file.rs

```diff
@@ -298,7 +298,29 @@ where
             elf::ET_REL => ObjectKind::Relocatable,
             elf::ET_EXEC => ObjectKind::Executable,
             // TODO: check for `DF_1_PIE`?
-            elf::ET_DYN => ObjectKind::Dynamic,
+            elf::ET_DYN => {
+                let mut is_pie = false;
+                let table: &SectionTable<'_, Elf, R> = self.elf_section_table();
+                if let Ok(Some(dyn_sec)) = table.dynamic(self.endian(), self.data()) {
+                    for v in dyn_sec.0 {
+                        let tag = v.tag32(self.endian());
+                        let val = v.val32(self.endian());
+                        if let Some(tag) = tag {
+                            if let Some(val) = val {
+                                if tag == DT_FLAGS_1 && val & DF_1_PIE == DF_1_PIE {
```

Comment:
LLVM only started setting this flag in 2020: https://reviews.llvm.org/D80872

Any ET_DYN file with non-zero e_entry and/or a PT_INTERP can be run as executable. And it is possible to make a file that is both a valid executable and a valid shared library. For example the dynamic linker on Linux generally is both.

Reply:
So maybe we should replace `ObjectKind` with methods like `is_executable` and `is_dynamic`?

Reply:
fiy this is how `file` detects PIE executable
https://stackoverflow.com/questions/34519521/why-does-gcc-create-a-shared-object-instead-of-an-executable-binary-according-to/55704865#55704865

I originally wanted to check the permissions but it's hard to implement it here (i need to know the path)